services/core-api/app/api/mines/tailings/models/tailings.py

```diff
@@ -20,10 +18,26 @@ class MineTailingsStorageFacility(AuditMixin, Base):
     consequence_classification_status_code = db.Column(db.String)
     has_itrb = db.Column(db.Boolean)
     tsf_operating_status_code = db.Column(db.String)
+    engineer_of_records = db.relationship(
+        'MinePartyAppointment',
+        lazy='select',
+        primaryjoin=
+        'and_(MinePartyAppointment.mine_tailings_storage_facility_guid == MineTailingsStorageFacility.mine_tailings_storage_facility_guid, MinePartyAppointment.mine_party_appt_type_code == "EOR", MinePartyAppointment.deleted_ind == False)',
+        order_by=
+        'nullsfirst(desc(MinePartyAppointment.start_date)), nullsfirst(desc(MinePartyAppointment.end_date))'
+    )
+
+    @hybrid_property
+    def engineer_of_record(self):
+        if self.engineer_of_records:
+            return self.engineer_of_records[0]
 
     def __repr__(self):
         return '<MineTailingsStorageFacility %r>' % self.mine_guid
 
+    def __str__(self):
```

Comment:
Did you create this for debugging purposes? It could be removed as we don't really declare these for any of our classes. No big deal, though.

Reply:
Yes, it was for sending to console the MineTailingStorageFacility object. It is removed :)